examples/pennylane_run_rotoselect.py

```diff
@@ -0,0 +1,429 @@
+# Therefore it would be useful to employ an
+# algorithm which learns a good circuit structure at fixed depth to minimize the cost function.
+#
+# Furthermore, PennyLane's optimizers perform automatic differentiation of quantum nodes by evaluating phase-shifted
+# expectation values using the quantum circuit itself.
+# The output of these calculations, the gradient, is used in optimization methods to minimize
+# the cost function. However,
+# there exists a technique to find the optimal parameters of a quantum circuit through phase-shifted evaluations,
+# without the need for calculating the gradient as an intermediate step (i.e., a gradient-free optimization).
+# It could be desirable, in some cases, to
+# take advantage of this.
+#
+#
+# The Rotoselect algorithm addresses the above two points.
+# It works by updating the parameters :math:`\boldsymbol{\theta}=\theta_1,\dots,\theta_D` and gate choices
+# :math:`\boldsymbol{P}=P_1...P_D`
+# one at a time according to a *closed-form expression for the optimal parameter value* :math:`\theta^{*}_d`
+# when the other parameters and gate choices are fixed:
+#
+# .. math::
+#
+#   \theta^{*}_d &= \underset{\theta_d}{\text{argmin}} \langle H \rangle_{\theta_d} \\
+#                &= -\frac{\pi}{2} - \text{arctan}\left(\frac{2\langle H \rangle_{\theta_d = 0} -
+#                \langle H \rangle_{\theta_d=\pi/2} - \langle H \rangle_{\theta_d=-\pi/2}}{\langle
+#                H \rangle_{\theta_d=\pi/2} -
+#                \langle H \rangle_{\theta_d=-\pi/2}}\right)
+#
+# The calculation makes use of 3 separate evaluations
+# of the expectation value :math:`\langle H \rangle_{\theta_d}` using the quantum circuit. Although
+# :math:`\langle H \rangle` is really a function of all parameters and gate choices
+# :math:`\boldsymbol{\theta}`, :math:`\boldsymbol{P}`, we
+# are fixing every parameter and gate choice apart from :math:`\theta_d` in this expression so we write it as
+# :math:`\langle H \rangle = \langle H \rangle_{\theta_d}`.
+# For each parameter in the quantum circuit, the algorithm proceeds by evaluating :math:`\theta^{*}_d`
+# for each choice of
+# gate :math:`R_d \in \{R_x,R_y,R_z\}` and selecting the gate which yields the minimum value of
+# :math:`\langle H \rangle`.
+#
+# Thus, one might expect the number of circuit evaluations required to be 9 for each parameter (3 for each gate
+# choice). However, since there is a 3-fold
+# degeneracy in the expectation value when :math:`\theta_d=0`,
+#
+# .. math:: R_x(0) = R_y(0) = R_z(0) = 1,
+#
+# the number of evaluations is reduced to 7.
+#
+# One cycle of the Rotosolve algorithm involves
+# iterating through every parameter and performing the calculation above.
+# This cycle is repeated for a fixed number of steps or until convergence. In this way, one could learn both
+# the optimal parameters and gate choices for a circuit ansatz. Next, we present an example of this algorithm
+# applied to a VQE Hamiltonian.
+#
+# VQE
+# ~~~
```

Comment:
This is the only heading in the 'background' section, and it reads a bit strangely. Anyway it can be expanded? i.e., 
```rest
Example VQE problem
~~~~~~~~~~~~~~~~~~
```
Also, I'm not sure if this fits in the background section - might make more sense to make this a top level heading?

Reply:
Okay I've changed this